test/e2e/util.go

```diff
@@ -482,35 +539,49 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 		return fmt.Errorf("Error creating replication controller: %v", err)
 	}
 
+	// Create a routine to query for the list of pods
+	stop := make(chan struct{})
+	go func(stop <-chan struct{}, n string, ns string, l labels.Selector) {
+		for {
+			select {
+			case <-stop:
+				return
+			default:
+				p, err := c.Pods(ns).List(l, fields.Everything())
+				if err != nil {
+					Logf("Warning: Failed to get pod list: %v", err)
+				} else {
+					podLists.Push(p.Items)
+				}
+				time.Sleep(1 * time.Second)
```

Comment:
We want as close as possible to 1 second granularity.  At our scale (23 noes, 100/node) I'm seeing the query take 1-1 1/2 seconds, sometimes a little more and i expect it would get worse the bigger the scale.  I'll take a look at watch